modules/compression/include/ips_jpg_at_model.cpp

```diff
@@ -0,0 +1,226 @@
+#include <systemc.h>
+
+/**
+ * @brief jpg_output module. Federico Cruz
+ * It takes the image and compresses it into jpeg format
+ * It is done in 4 parts:
+ * 1. Divides the image in 8x8 pixel blocks; for 8-bit grayscale images the a level shift is done by substracting 128 from each pixel.
+ * 2. Discrete Cosine Transform (DCT) of the 8x8 image.
+ * 3. Each transformed 8x8 block is divided by a quantization value for each block entry.
+ * 4. Each quantized 8x8 block is reordered by a Zig-Zag sequence into a array of size 64.
+ * *5. Entropy compression by variable length encoding (huffman). Used to maximize compression. Not implemented here.
+ */
+#define PI 3.1415926535897932384626433832795
+#define Block_rows 8
+#define Block_cols 8
+
+SC_MODULE (jpg_output) {
+  
+  //input signals
+  sc_in<sc_int<32> > PixelValue_signal;
+  sc_in<sc_int<32> > row_signal;
+  sc_in<sc_int<32> > col_signal;
+  
+  //output signals
+  sc_out<sc_int<8> > Element_signal;
+  sc_in<sc_int<32> > index_signal;
+  
+  //compression signals
+  sc_out<sc_int<32> > output_size_signal;
+  
+  //-----------Internal variables-------------------
+  //const int Block_rows = 8;
+  //const int Block_cols = 8;
+  double* image;
+  int image_rows = 480;
+  int image_cols = 640;
+  signed char EOB = 127; // end of block
+  
+  int quantificator[8][8] = { // quantization table
+  {16,11,10,16,24,40,51,61},
+  {12,12,14,19,26,58,60,55},
+  {14,13,16,24,40,57,69,56},
+  {14,17,22,29,51,87,80,62},
+  {18,22,37,56,68,109,103,77},
+  {24,35,55,64,81,104,113,92},
+  {49,64,78,87,103,121,120,101},
+  {72,92,95,98,112,100,103,99}};
+
+  int zigzag_index[64]={ // zigzag table
+  0,1,5,6,14,15,27,28,
+  2,4,7,13,16,26,29,42,
+  3,8,12,17,25,30,41,43,
+  9,11,18,24,31,40,44,53,
+  10,19,23,32,39,45,52,54,
+  20,22,33,38,46,51,55,60,
+  21,34,37,47,50,56,59,61,
+  35,36,48,49,57,58,62,63};
+
+  sc_event starter_event;
+  sc_event input_event;
+  sc_event output_event;
+  sc_event compression_event;
+
+  // Constructor for compressor
+  SC_HAS_PROCESS(jpg_output);
+    jpg_output(sc_module_name jpg_compressor): sc_module(jpg_compressor){
+      image = new double[image_rows*image_cols];
+      //initialize the image matrix to avoid nan
+      for(int i=0; i<(image_rows*image_cols);i++){
+		image[i]=0;
+	  }
+	  SC_THREAD(starter_operation);
+	  SC_THREAD(input_operation);
+	  SC_THREAD(output_operation);
+	  SC_THREAD(compression_operation);
+  } // End of Constructor
+
+  //------------Code Starts Here-------------------------
+  void Starter() {
```

Comment:
Use consistent notation for methods and variable names like camelCase, snake_case, or PascalCase.